src/features/recieve/AddressBlock.tsx

```diff
@@ -0,0 +1,226 @@
+import cx from 'classnames'
+import { clipboard, shell } from 'electron'
+import QRCode from 'qrcode.react'
+import React, { Component } from 'react'
+import {
+  AccordionItem,
+  AccordionItemButton,
+  AccordionItemHeading,
+  AccordionItemPanel,
+} from 'react-accessible-accordion'
+
+import cstyles from '../../legacy/components/Common.module.css'
+import Utils from '../../legacy/utils/utils'
+import styles from './Receive.module.css'
+
+export interface IAddressBlockProps {
+  currencyName: string
+  viewKey: string
+  fetchAndSetSingleViewKey(add: string): void
+
+  privateKey?: string
+  hidePrivKey(): void
+  fetchAndSetSinglePrivKey(add: string, type?: string): void
+
+  addressBalance: {
+    address: string
+    balance: number
+  }
+  label?: string
+
+  // deprecated, see Utils.getPslToUsdString
+  pslPrice: number
+}
+export interface IAddressBlockState {
+  copiedAddr: boolean
+  copiedPrivKey: '' | 'loading' | 'done'
+}
+
+export class AddressBlock extends Component<
```

Comment:
Could we please have this as a functional component?

Reply:
Migrated to functional component